packages/core-ethereum/src/commitment.ts

```diff
@@ -43,7 +44,7 @@ export class Commitment {
     let result = await recoverIteratedHash(
       hash.serialize(),
       hashFunction,
-      async (x) => await this.searchDBFor(x),
+      this.searchDBFor.bind(this),
```

Comment:
am more in favor of the previous version, I try to avoid using `.bind` whenever possible

Reply:
I don't think so. Using a wrapper function to implicitly bind a function to a context don't look like a good solution to me. See also https://javascript.info/bind why using .bind() is more precise than a wrapper function.

Please give arguments why .bind() should be prevented!

Reply:
> Using a wrapper function to implicitly bind a function to a context

u don't have to bind a function into a context

> why using .bind() is more precise than a wrapper function

ur argument here makes no sense

I simply find `.bind` less readable, feel free to disagree, I don't want to go in-depth on this u can keep it

Reply:
> Using a wrapper function to implicitly bind a function to a context

This **is** the semantics that happens when using a wrapper function, just less explicit